IPython/terminal/interactiveshell.py

```diff
@@ -289,30 +305,15 @@ def _make_style_from_name_or_cls(self, name_or_cls):
                 # and a light background, because we can't tell what the terminal
                 # looks like. These tweaks to the default theme help with that.
                 style_cls = get_style_by_name('default')
-                style_overrides.update({
-                    Token.Number: '#007700',
-                    Token.Operator: 'noinherit',
-                    Token.String: '#BB6622',
-                    Token.Name.Function: '#2080D0',
-                    Token.Name.Class: 'bold #2080D0',
-                    Token.Name.Namespace: 'bold #2080D0',
-                    Token.Prompt: '#009900',
-                    Token.PromptNum: '#00ff00 bold',
-                    Token.OutPrompt: '#990000',
-                    Token.OutPromptNum: '#ff0000 bold',
-                })
+                style_overrides = _style_overrides_neutral
 
                 # Hack: Due to limited color support on the Windows console
                 # the prompt colors will be wrong without this
                 if os.name == 'nt':
-                    style_overrides.update({
-                        Token.Prompt: '#ansidarkgreen',
-                        Token.PromptNum: '#ansigreen bold',
-                        Token.OutPrompt: '#ansidarkred',
-                        Token.OutPromptNum: '#ansired bold',
-                    })
-            elif legacy =='nocolor':
-                style_cls=_NoStyle
```

Comment:
That seem strait forward here, but actually the semantics are different, and I even believe some other branches in the code are wrong, In particular:

```
style_overrides = _style_overrides_linux
``` 

and 

```
style_overrides = _style_overrides_light_bg
```

Indeed, later on we do `style_overrides.update(self.highlighting_style_overrides)`, so if you assign something to `style_override` instead of using `update(...)` you will mutate the original. `_style_overrides_light_bg`, `_style_overrides_linux` and with your patch `_style_overrides_windows` which is surprising. 

So yes to deport them as global variable, but we need to be sure of what update() vs  assing will do.

Reply:
So, does that mean the `_style_overrides_windows` part is not going to work?